src/state/events.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+namespace genie {
+
+enum Sound_t {
+  SOUND_NO_MATCH = -1,
+  SOUND_MATCH = 0,
+  SOUND_NEWS_INTRO = 1,
+  SOUND_ALARM_CLOCK_ELAPSED = 2,
+};
+
+struct AudioFrame {
+  int16_t *samples;
+  size_t length;
+
+  AudioFrame(size_t len) : samples(new int16_t[len]), length(len) {}
+  ~AudioFrame() { delete[] samples; }
+
+  AudioFrame(const AudioFrame &) = delete;
+  AudioFrame &operator=(const AudioFrame &) = delete;
+
+  AudioFrame(AudioFrame &&other)
+      : samples(other.samples), length(other.length) {
+    other.samples = nullptr;
+    other.length = 0;
+  }
+};
+
+namespace state {
+namespace events {
+
+struct Event {};
+
+struct Wake : Event {};
+
+struct InputFrame : Event {
+  AudioFrame *frame;
+
+  InputFrame(AudioFrame *frame);
+  ~InputFrame();
+};
+
+struct InputDone : Event {};
+
+struct InputNotDetected : Event {};
+
+struct InputTimeout : Event {};
+
+// ### Conversation Events ###
+
+struct TextMessage : Event {
+  gchar *text;
```

Comment:
Would be simpler for these to be std::string, in all events. You'd save a destructor definition and would be sure there are no leaks.

Reply:
Yeah I like that better. Wasn't sure if we were against `std::string` for some reason 'cause I haven't seen it used anywhere in the codebase

Reply:
Fixed in 1dd6d2cac536547cdfcc4ec918eac1ef8651b5b1